core/trino-spi/src/main/java/io/trino/spi/expression/ConnectorExpression.java

```diff
@@ -13,26 +13,34 @@
  */
 package io.trino.spi.expression;
 
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
 import io.trino.spi.type.Type;
 
 import java.util.List;
 
 import static java.util.Objects.requireNonNull;
 
+@JsonTypeInfo(use = JsonTypeInfo.Id.CLASS)
```

Comment:
Don't you need to enumerate known subclasses as in https://github.com/trinodb/trino/blob/36ff8b5a3d69e244ac7eb7d17f63a12458f41f83/core/trino-main/src/main/java/io/trino/sql/planner/plan/PlanNode.java#L28-L69 ?

Reply:
no, see https://github.com/trinodb/trino/pull/11447#issuecomment-1066905919